### ISSUE TITLE
Router cant uncompress files

### ISSUE THREAD

**Issue by Jiaocz** [acecilia/OpenWRTInvasion#31]
When it uploaded config files (payload.tar.gz), it said "code":"1629:, msg is Error uncompressingm, the fies may damaged.
![image](https://user-images.githubusercontent.com/14857984/92240969-087a0e80-eef0-11ea-8000-ce68e3a46b22.png)

**Comment by acecilia** [comment]
I think that message is received every time, the exploit works anyway

**Comment by Jiaocz** [comment]
I found the reason, it is because that in the speedtest, some url can be accessed in my area, but if I replace some of these, then the message will become "Error speedtesting" but not Script executed.

**Comment by acecilia** [comment]
Make sure your network configuration is as simple as possible, that you are configuring the right IPs and that connection between your computer and the router is happening

**Comment by hstr0100** [comment]
Not sure where to put this comment, but it should help whoever has the same problem as me:

On my 4A, versions 0.0.2 and 0.0.3 of the exploit did not work. the commands did get executed on the router but the telnet daemon did not work.
I managed to gain shell access on the firmware 2.28.132 by using netcat (version 0.0.1 of this exploit).

From there, I manually executed the exploit command and got the output "curl (51) error"
This means that for some reason my router wasn't able to verify the SSL certificate of the busybox download site, so from there everything failed.
by slapping --insecure to the curl command string I was able to successfully run the exploit v0.0.3.

So if you run into the same problem as me, append `--insecure` to https://github.com/acecilia/OpenWRTInvasion/blob/master/script.sh#L17 and have fun.
Some may critique the fact that we're ignoring the SSL certificate there, and that could be potentially dangerous, but this is no worse than "curl-ing" an insecure http:// url anyway, and there's not much to do to fix this problem anyway aside from including busybox into the payload I guess.

By doing this you also don't have to reset your router nor anything. it seems that this is the cause of most of the issues previously related here. so I hope this helps so we can get more people onboard custom firmwares for this router.
